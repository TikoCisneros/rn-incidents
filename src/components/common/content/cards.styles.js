import { StyleSheet } from 'react-native';
import Palette from '../../../styles/palette';

const boxesStyles = StyleSheet.create({
  card: {
    borderRadius: 8,
    borderStyle: 'solid',
    borderWidth: 0.2,
    borderColor: Palette.grayLight,
    shadowColor: Palette.grayPrincipal,
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
    elevation: 3,
    backgroundColor: Palette.white,
    marginHorizontal: 2,
  },
});

export default boxesStyles;
