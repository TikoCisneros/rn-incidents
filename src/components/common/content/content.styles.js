import { css } from 'styled-components/native';

const contentStyles = css`
  flex: 1;
  padding: 0 5%;
`;

const fullScreenStyles = css`
  flex: 1;
`;

export { contentStyles, fullScreenStyles };
