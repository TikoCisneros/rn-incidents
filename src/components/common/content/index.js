export * from './content';
export * from './cards';
