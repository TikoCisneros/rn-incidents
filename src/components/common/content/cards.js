import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import PropTypes from 'prop-types';

import { isFunction } from '../../../common/util';
import boxesStyles from './cards.styles';

const Card = ({ style, onPress, children }) => {
  const Container = isFunction(onPress) ? TouchableOpacity : View;

  return (
    <Container style={[boxesStyles.card, style]} onPress={onPress}>
      {children}
    </Container>
  );
};

Card.propTypes = {
  onPress: PropTypes.func,
};

Card.defaultProps = {
  onPress: undefined,
};

// eslint-disable-next-line import/prefer-default-export
export { Card };
