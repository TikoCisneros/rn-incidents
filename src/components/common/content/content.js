import React from 'react';
import styled from 'styled-components/native';
import LinearGradient from 'react-native-linear-gradient';
import { contentStyles } from './content.styles';

const Content = styled.View`
  ${contentStyles}
`;

const ScrollableContent = styled.ScrollView`
  ${contentStyles}
`;

const LinearBg = styled(LinearGradient)`
  ${contentStyles}
`;

// eslint-disable-next-line react/prop-types
const LinearBgContent = ({ colors, children }) => (
  <LinearBg colors={colors} start={{ x: 0, y: 0 }} end={{ x: 1, y: 1 }}>
    {children}
  </LinearBg>
);

export { Content, ScrollableContent, LinearBgContent };
