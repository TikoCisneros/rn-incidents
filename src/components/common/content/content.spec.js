import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Content, ScrollableContent, LinearBgContent } from './content';

describe('Content components', () => {
  it('should render Content successfully', () => {
    const component = <Content />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render ScrollableContent successfully', () => {
    const component = <ScrollableContent />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render LinearBgContent successfully', () => {
    const component = <LinearBgContent colors={['purple', 'white']} />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
