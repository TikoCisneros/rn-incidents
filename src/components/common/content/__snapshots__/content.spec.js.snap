// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content components should render Content successfully 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": "5%",
        "paddingRight": "5%",
        "paddingTop": 0,
      },
    ]
  }
/>
`;

exports[`Content components should render LinearBgContent successfully 1`] = `
<BVLinearGradient
  colors={
    Array [
      4286578816,
      4294967295,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": "5%",
        "paddingRight": "5%",
        "paddingTop": 0,
      },
    ]
  }
/>
`;

exports[`Content components should render ScrollableContent successfully 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": "5%",
        "paddingRight": "5%",
        "paddingTop": 0,
      },
    ]
  }
>
  <View />
</RCTScrollView>
`;
